python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -148,6 +148,22 @@ def get_lsst_photometric_uncertainties(self):
                                                  'lsstBandpassDict')
 
 
+    def calculateVisibility(self, magFilter, m5, sigma=0.12):
+        """
+        Calculate the probability of detecting a particular source.
+
+        @ param [in] magFilter is the magnitude of the object in the given filter.
+
+        @ param [in] obs_metadata is the observation metadata for a given visit (to determine m5).
+
+        @ param [out] visibility (0/1).
+        """
+        if len(magFilter) == 0:
+            return numpy.array([])
+        completeness = 1.0 / (1 + numpy.exp((magFilter - m5)/sigma))
+        probability = numpy.random.random_sample(len(magFilter))
```

Comment:
Okay. 
If I just want it to be generally reproducible, I just need to add the option to specify a seed here, and then somewhere I need to specify the seed and pass it in. 
If I am going to say "it's reproducible for nominal catalogs", I only have to set the random seed once per visit/instance catalog (because I'm calling this for all objects in the instance catalog) and then assume that 'nominal catalogs' came back in the same objid order (I know that's not really the same  .. unless there's a sort by objid somewhere, but it's probably close enough for me for now). So I guess I'll just set it when I call this, using the obshistid maybe.